bittensor/_dataset/dataset_impl.py

```diff
@@ -1,536 +1,490 @@
-""" Implementation for the dataset and GenesisTextDataset class, which handles dataloading from ipfs
+""" Implementation of Axon, services Forward and Backward requests from other neurons.
 """
 # The MIT License (MIT)
 # Copyright © 2021 Yuma Rao
+# Copyright © 2022 Opentensor Foundation
 
-# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
-# documentation files (the “Software”), to deal in the Software without restriction, including without limitation
-# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+# documentation files (the “Software”), to deal in the Software without restriction, including without limitation 
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, 
 # and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 
-# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of 
 # the Software.
 
 # THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
-# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
-# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION 
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
 # DEALINGS IN THE SOFTWARE.
 
-import concurrent
-import json
-import os
-import random
-import time
-from multiprocessing import cpu_count
-from typing import Union
 
-import requests
 import torch
-from loguru import logger
-from requests.adapters import HTTPAdapter
-from requests.packages.urllib3.util.retry import Retry
+import concurrent.futures
+import time
+import psutil
+import sys
+import random
+import argparse
 from torch.utils.data.dataloader import DataLoader
-
+from tqdm import tqdm
 import bittensor
+import glob
+from queue import Queue
+import numpy as np
+import asyncio
+import aiohttp
+import json
+import os
+import threading
+import asyncio
+import nest_asyncio
+from loguru import logger
+##########################
+##### Get args ###########
+##########################
+from typing import *
```

Comment:
i've got a lot of unused imports flagged here, could you tidy them? (`pylint` should flag them easy and you can use `isort` to do PEP8 sorting super quick)

Reply:
cleaned it up by removing unused libraries